solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+// TODO: Documentation
+// TODO: Unit tests
+library Relay {
+    struct Request {
+        uint256 id;
+        uint256 startBlock;
+        uint256 groupIndex;
+        bytes previousEntry;
+    }
```

Comment:
Regardless if we decide to use a library for `Relay`, we need to revisit types used in this struct. `uint256` for `groupIndex` feels too much, I'd also argue `id` and `startBlock` can probably be reduced, see https://fravoll.github.io/solidity-patterns/tight_variable_packing.html.